psdisk/psdisk.c

```diff
@@ -359,227 +394,158 @@ static int psdisk_updatePartsList(ptable_partition_t *pHeaders, uint32_t pCnt)
+	struct plist_node_t *node;
+	const char *nptr;
+	char *endptr;
+	int i;
 
-	/* Get data from optarg and separete them */
-	if ((argsStr = (char *)malloc(strlen(optarg) + 1)) == NULL) {
-		fprintf(stderr,"Cannot allocate memory, err: %s.\n", strerror(errno));
+	node = calloc(1, sizeof(struct plist_node_t));
+	if (node == NULL) {
+		fprintf(stderr, "Cannot allocate memory, err: %s.\n", strerror(errno));
 		return -1;
 	}
 
-	do {
-		strcpy(argsStr, optarg);
-		if ((opts[0] = strtok(argsStr, delimiter)) == NULL) {
-			fprintf(stderr,"Invalid number of arguments.\n");
-			break;
-		}
-
-		for (i = 1; i < MAX_PARTITION_OPTS; ++i) {
-			if ((opts[i] = strtok(NULL, delimiter)) == NULL) {
-				fprintf(stderr,"Invalid number of arguments.\n");
-				ret = 1;
-				break;
-			}
-		}
-
-		if (ret)
-			break;
-
-		if ((pNode = (struct plist_node_t *)calloc(1, sizeof(struct plist_node_t))) == NULL) {
-			fprintf(stderr,"Cannot allocate memory, err: %s.\n", strerror(errno));
-			break;
-		}
-
-		/* Parse partition name */
-		for (i = 0; i < strlen(opts[0]); ++i) {
-			if (!isalnum(opts[0][i])) {
-				fprintf(stderr,"Invalid partition name - %s.\n", opts[0]);
-				ret = 1;
-				break;
-			}
-		}
-
-		if (ret)
-			break;
-
-		if (strlen(opts[0]) >= sizeof(pNode->pHeader.name)) {
-			fprintf(stderr,"Invalid partition name - %s.\n", opts[0]);
-			break;
-		}
-		else {
-			strncpy((char *)pNode->pHeader.name, opts[0], sizeof(pNode->pHeader.name));
-		}
-
-		if (((pNode->pHeader.offset = strtoul(opts[1], &endptr, 0)) == 0 && strlen(endptr) != 0) || (pNode->pHeader.offset != 0 && strlen(endptr) != 0)) {
-			fprintf(stderr,"Partition %s has invalid offset %s.\n", pNode->pHeader.name, endptr);
-			break;
-		}
-
-		if ((pNode->pHeader.size = strtoul(opts[2], &endptr, 0)) == 0 || strlen(endptr) != 0) {
-			fprintf(stderr,"Partition %s has invalid size %s.\n", pNode->pHeader.name, endptr);
-			break;
-		}
-
-		if ((pNode->pHeader.type = strtoul(opts[3], &endptr, 0)) == 0 || strlen(endptr) != 0) {
-			fprintf(stderr,"Partition %s has invalid type %s.\n", pNode->pHeader.name, endptr);
+	/* Parse partition name */
+	for (i = 0; (i < sizeof(node->part.name)) && (arg[i] != '\0'); i++) {
+		if (!isalnum(arg[i])) {
 			break;
 		}
+	}
 
-		pNode->status = part_save;
-
-		LIST_INSERT_HEAD(&psdisk_common.list, pNode, ptrs);
-		++psdisk_common.pCnt;
-
-		free(argsStr);
-		return 0;
-
-	} while(0);
-
-	free(pNode);
-	free(argsStr);
-
-	return -1;
-}
-
-
-static int psdisk_parseMem(void)
-{
-	char *argsStr, *endptr;
-	const char *delimiter = ",";
-
-	const int MAX_MEM_OPTS = 2;
-	char *opts[MAX_MEM_OPTS];
-
-	/* Get data from optarg and separete them */
-	if ((argsStr = malloc(strlen(optarg) + 1)) == NULL) {
-		fprintf(stderr,"Cannot allocate memory, err: %s.\n", strerror(errno));
+	if ((i == 0) || (i >= sizeof(node->part.name)) || (arg[i] != ',')) {
+		fprintf(stderr, "Invalid partition name - %s.\n", arg);
+		free(node);
 		return -1;
 	}
+	strncpy((char *)node->part.name, arg, i);
```

Comment:
For safety, you can add this ( despite the `calloc()` used above).
```suggestion
	strncpy((char *)node->part.name, arg, i);
	node->part.name[i] = '\0';
```

Reply:
Thanks, added in the next force-push.